Magefile.go

```diff
@@ -0,0 +1,133 @@
+//go:build mage
+// +build mage
+
+// Package main provides Mage build targets for linting, running tests, and
+// orchestrating a Docker‑Compose test stack.  All public functions are
+// exported targets; every docstring is wrapped to 80 columns so that the file
+// remains readable in split‑screen editors.
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/exec"
+	"strings"
+
+	"github.com/magefile/mage/mg"
+	"github.com/magefile/mage/sh"
+)
+
+const (
+	composeFile = "./acceptance/docker-compose.yaml" // path to Compose file
+	projectName = "tests"                            // Compose project name
+)
+
+// composeEnv builds the environment passed to every docker‑compose command. It
+// injects the current user‑ID so containers can create files with the correct
+// ownership, enables Terraform acceptance tests, and forces Compose to emit
+// plain progress output (useful in CI logs).
+func composeEnv() map[string]string {
+	uidBytes, _ := exec.Command("id", "-u").Output()
+	return map[string]string{
+		"ZITADEL_DEV_UID":  strings.TrimSpace(string(uidBytes)),
+		"TF_ACC":           "1",
+		"COMPOSE_PROGRESS": "plain",
+	}
+}
+
+// runTests brings the stack up, executes `go test`, and always tears the stack
+// down.  Extra arguments (e.g. –coverprofile flags) are forwarded to the `go
+// test` invocation so that callers can run both plain and coverage runs with
+// the same orchestration logic.
+func runTests(ctx context.Context, extraArgs ...string) error {
+	if err := Up(); err != nil {
+		return err
+	}
+	defer Down()
+
+	args := append([]string{"test"}, extraArgs...)
+	args = append(args, "./...")
+	return sh.RunV("go", args...)
+}
+
+// Test is a Mage namespace.  `mage test:default` runs the normal test suite,
+// while `mage test:cover` writes a coverage profile.
+type Test mg.Namespace
+
+// Default executes the full acceptance test suite against the Compose stack.
+func (Test) Default(ctx context.Context) error {
+	return runTests(ctx)
+}
+
+// Cover runs the same suite but saves coverage data to profile.cov so that CI
+// can upload the report.
+func (Test) Cover(ctx context.Context) error {
+	return runTests(ctx, "-coverprofile=profile.cov")
+}
+
+// Lint invokes golangci‑lint if it is available; otherwise it falls back to
+// `go vet` so that developers without the linter installed can still run the
+// target locally without errors.
+func Lint() error {
+	if err := sh.Run("golangci-lint", "run", "./..."); err == nil {
+		return nil
+	}
+	fmt.Println("golangci-lint not found – falling back to `go vet`")
+	return sh.RunV("go", "vet", "./...")
```

Comment:
Let's try to have reproducible results by requiring a specific version of golangci-lint pinned in go.mod?
https://go.dev/doc/go1.24#tools